src/machine/machine_nrf52xxx.go

```diff
@@ -176,39 +218,28 @@ func (a ADC) Get() uint16 {
 // SPI on the NRF.
 type SPI struct {
 	Bus *nrf.SPIM_Type
-	buf *[1]byte // 1-byte buffer for the Transfer method
 }
 
-// There are 3 SPI interfaces on the NRF528xx.
-var (
-	SPI0 = SPI{Bus: nrf.SPIM0, buf: new([1]byte)}
-	SPI1 = SPI{Bus: nrf.SPIM1, buf: new([1]byte)}
-	SPI2 = SPI{Bus: nrf.SPIM2, buf: new([1]byte)}
-)
-
 // SPIConfig is used to store config info for SPI.
 type SPIConfig struct {
 	Frequency uint32
 	SCK       Pin
 	SDO       Pin
 	SDI       Pin
 	LSBFirst  bool
-	Mode      uint8
+	Mode      SPIMode
 }
 
-// Configure is intended to setup the SPI interface.
-func (spi SPI) Configure(config SPIConfig) error {
+// Configure is intended to set up the SPI interface.
+func (spi *SPI) Configure(config SPIConfig) error {
```

Comment:
The pointer receiver is not a change I'm against, in fact I think it may make sense to change it. I'd be curious as to why you are changing it :)

Reply:
- Well the SPI interfaces are created as singletons, so why should we copy it on its receivers.
- One would usually expect to configure the very instance where `Configure` is attached to and not a copy (that gets thrown away. Technically we configure the hardware not the instance but still it feels odd and as I said counter intuitive.